Fail with GHC 9.2 and Win32 2.12

A build log is:

```
Failed to build Glob-0.10.1.
Build log (
C:\Users\kazuki\AppData\Roaming\cabal\logs\ghc-9.2.1\Glob-0.10.1-ba443ecd253f1ed0cdc0f5bb4fab897565635639.log
):
Preprocessing library for Glob-0.10.1..
Building library for Glob-0.10.1..
[1 of 7] Compiling System.FilePath.Glob.Utils ( System\FilePath\Glob\Utils.hs, dist\build\System\FilePath\Glob\Utils.o )

System\FilePath\Glob\Utils.hs:29:28: error:
    Module ‘System.Win32.File’ does not export ‘c_GetFileAttributes’
   |
29 | import System.Win32.File  (c_GetFileAttributes, fILE_ATTRIBUTE_DIRECTORY)
   |                            ^^^^^^^^^^^^^^^^^^^
cabal: Failed to build Glob-0.10.1. See the build log above for details.
```

[Win32 2.12's change log](https://github.com/haskell/win32/blob/master/changelog.md#21200-march-2021) says:

> Add export lists to all modules, hiding numerous internal c_ bindings.

--- a/System/FilePath/Glob/Utils.hs
+++ b/System/FilePath/Glob/Utils.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
 -- File created: 2008-10-10 13:40:35
 
 module System.FilePath.Glob.Utils
@@ -26,7 +27,8 @@ import System.IO.Unsafe (unsafeInterleaveIO)
 #if mingw32_HOST_OS
 import Data.Bits          ((.&.))
 import System.Win32.Types (withTString)
-import System.Win32.File  (c_GetFileAttributes, fILE_ATTRIBUTE_DIRECTORY)
+import System.Win32.File  (FileAttributeOrFlag, fILE_ATTRIBUTE_DIRECTORY)
+import System.Win32.String (LPCTSTR)
 #else
 import Foreign.C.String      (withCString)
 import Foreign.Marshal.Alloc (allocaBytes)
@@ -123,6 +125,10 @@ doesDirectoryExist s =
             if st == 0
                then fmap s_isdir (st_mode p)
                else return False
+#endif
+
+#if mingw32_HOST_OS
+foreign import ccall unsafe "windows.h GetFileAttributesW" c_GetFileAttributes :: LPCTSTR -> IO FileAttributeOrFlag
 #endif
 
 getRecursiveContents :: FilePath -> IO (DList FilePath)

--- a/System/FilePath/Glob/Utils.hs
+++ b/System/FilePath/Glob/Utils.hs
@@ -128,7 +128,13 @@ doesDirectoryExist s =
 #endif
 
 #if mingw32_HOST_OS
+#if defined(i386_HOST_ARCH)
+foreign import stdcall unsafe "windows.h GetFileAttributesW" c_GetFileAttributes :: LPCTSTR -> IO FileAttributeOrFlag
+#elif defined(x86_64_HOST_ARCH)
 foreign import ccall unsafe "windows.h GetFileAttributesW" c_GetFileAttributes :: LPCTSTR -> IO FileAttributeOrFlag
+#else
+#error Unknown mingw32 arch
+#endif
 #endif
 
 getRecursiveContents :: FilePath -> IO (DList FilePath)

